cranelift/entity/src/sparse.rs

```diff
@@ -203,6 +204,29 @@ where
     }
 }
 
+impl<K, V> Default for SparseMap<K, V>
+where
+    K: EntityRef,
+    V: SparseMapValue<K>,
+{
+    fn default() -> SparseMap<K, V> {
+        SparseMap::new()
+    }
+}
+
+impl<K, V> fmt::Debug for SparseMap<K, V>
+where
+    K: EntityRef + fmt::Debug,
+    V: SparseMapValue<K> + fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SparseMap")
+            .field("sparse", &self.sparse)
+            .field("dense", &self.dense)
+            .finish()
```

Comment:
The sparse and dense fields are not exposed to the user and rather useless unless you know how the implementation lf SparseMap works. I would much rather prefer the Debug impl being like that of HashMap showing the logical mapping from keys to values. So for example `{ v0: "a", v2: "b" }` for a `SparseMap<Value, String>`. This can be done using something like `f.debug_map().entries(entries).finish()` where `entries` is an iterator over the map entries.

Reply:
makes sense! updated.